manifests/helm-watcher/helm_watcher_role.yaml.tpl

```diff
@@ -0,0 +1,30 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: helm-watcher-role
+  namespace: {{ .Namespace }}
+rules:
+  - apiGroups:
+      - helm.watcher
```

Comment:
shouldn't this be `source.toolkit.fluxcd.io`?

Reply:
nope

Reply:
This is the apiGroup of the helm watcher which is defined as helm.watcher.... uh, somewhere. :D I remember defining it.

Reply:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#policyrule-v1-rbac-authorization-k8s-io
```
apiGroups
string array	APIGroups is the name of the APIGroup that contains the resources. If multiple API groups are specified, any action requested against one of the enumerated resources in any API group will be allowed.
```

doesn't the resource belong to the `source.toolkit.fluxcd.io` API group?

Reply:
The resource in this case is the watcher. It's the `resources` that is the important one.

Reply:
isn't the resource the `HelmRepository`? The same as we do for the regular `Role` for wego-app:
```
  - apiGroups: ["helm.toolkit.fluxcd.io"]
    resources: [ "helmreleases" ]
    verbs: [ "*" ]
```

https://github.com/weaveworks/weave-gitops/blob/main/manifests/wego-app/role.yaml.tpl

Reply:
nope, it's plural, `helmrepositories`. 

However, the API group might be different though. For now, that should be the wego-app, that's true.

Reply:
sorry I mean't `helmrepositories`. I'm still pretty adamant that the API group isn't `helm.watcher` though. But I could be wrong still
```
  - apiGroups: ["source.toolkit.fluxcd.io"]
    resources: [ "helmrepositories" ]
    verbs: [ "*" ]
```

https://github.com/weaveworks/weave-gitops/blob/1fd04684d2f076e4068670d42ff11535e7db9f0e/manifests/wego-app/role.yaml.tpl#L19-L21

Reply:
Ah. Right. Yes. True

Reply:
@aclevername is right, the watcher requires permissions on objects to watch...so, you're saying that it has permissions to watch watch objects in the "helm.watcher" API group...which isn't a thing...

(Even if it was, it needs permissions to watch flux objects)

Reply:
@bigkevmcd yeah we discussed this. I was still in the land where I generated my sample project using kubebuilder. :) I had the wrong context in my head.